dpnp/dpnp_algo/dpnp_algo_mathematical.pyx

```diff
@@ -344,6 +320,24 @@ cpdef dparray dpnp_multiply(dparray x1, x2):
         return call_fptr_2in_1out(DPNP_FN_MULTIPLY, x1, x2, x1.shape)
 
 
+cpdef dparray dpnp_nancumprod(dparray x1):
+
+    for i in range(x1.size):
```

Comment:
The same comment as above, this is wrong logic, do not modify given array, just check it.